src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+import { getUser } from '../../api';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo | null;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const todoUserId = selectedTodo?.userId;
+
+    if (todoUserId) {
+      setLoading(true);
+      getUser(todoUserId)
+        .then(setSelectedUser)
+        .finally(() => {
+          setLoading(false);
+        });
```

Comment:
Consider adding error handling to the `getUser(todoUserId)` promise chain. This will help manage any potential errors that occur during the fetching of user data. You can use a `.catch()` block to handle errors.